pkg/endpoint/bpf.go

```diff
@@ -1146,25 +1156,75 @@ func (e *Endpoint) addPolicyKey(keyToAdd policy.Key, entry policy.MapStateEntry,
 
 // ApplyPolicyMapChanges updates the Endpoint's PolicyMap with the changes
 // that have accumulated for the PolicyMap via various outside events (e.g.,
-// identities added / deleted).
-// 'proxyWaitGroup' may not be nil.
-func (e *Endpoint) ApplyPolicyMapChanges(proxyWaitGroup *completion.WaitGroup) error {
-	if err := e.lockAlive(); err != nil {
-		return err
-	}
-	defer e.unlock()
+// identities added / deleted) by triggering the incremental-policy-map controller.
+// Returns a channel that is closed when the changes are applied. The channel
+// is closed even if errors occurred. The controller will retry the changes.
+func (e *Endpoint) ApplyPolicyMapChanges() channels.DoneChan {
+	return e.updateIncrementalPolicyMapController()
+}
 
-	e.PolicyDebug(nil, "ApplyPolicyMapChanges")
+// updateIncrementalPolicyMapController creates or triggers a controller to reconcile
+// pending policy map changes.
+func (e *Endpoint) updateIncrementalPolicyMapController() channels.DoneChan {
+	ch := make(chan struct{})
```

Comment:
Does this work? What happens if multiple callers get "batched" in to the same update? Won't one of the channels be lost?

I strongly strongly strongly recommend using TriggerController, not UpdateController.